packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -84,11 +87,41 @@ export function createAlgoliaInsightsPlugin(
   options: CreateAlgoliaInsightsPluginParams
 ): AutocompletePlugin<any, undefined> {
   const {
-    insightsClient,
+    insightsClient: providedInsightsClient,
     onItemsChange,
     onSelect: onSelectEvent,
     onActive: onActiveEvent,
   } = getOptions(options);
+  let insightsClient = providedInsightsClient as InsightsClient;
+
+  if (!providedInsightsClient) {
+    safelyRunOnBrowser(({ window }) => {
+      const pointer = window.AlgoliaAnalyticsObject || 'aa';
+
+      if (typeof pointer === 'string') {
+        insightsClient = window[pointer];
+      }
+
+      if (!insightsClient) {
+        window.AlgoliaAnalyticsObject = pointer;
+
+        if (!window[pointer]) {
+          window[pointer] = (...args: any[]) => {
+            if (!window[pointer].queue) {
+              window[pointer].queue = [];
+            }
+
+            window[pointer].queue.push(args);
+          };
+        }
```

Comment:
Should we now attach the version to the function queue?

```suggestion
          window[pointer].version = ALGOLIA_INSIGHTS_VERSION;
        }
```

Reply:
Yes! Added it in [6a51c83](https://github.com/algolia/autocomplete/pull/1106/commits/6a51c8352bac386427f1970271bb911b313be198) with tests.